qa/rpc-tests/reindex.py

```diff
@@ -21,13 +22,22 @@ def setup_network(self):
         self.is_network_split = False
         self.nodes.append(start_node(0, self.options.tmpdir))
 
-    def run_test(self):
+    def reindex(self, justchainstate=False):
         self.nodes[0].generate(3)
+        blockcount = self.nodes[0].getblockcount()
         stop_node(self.nodes[0], 0)
         wait_bitcoinds()
-        self.nodes[0]=start_node(0, self.options.tmpdir, ["-debug", "-reindex", "-checkblockindex=1"])
-        assert_equal(self.nodes[0].getblockcount(), 3)
+        self.nodes[0]=start_node(0, self.options.tmpdir, ["-debug", "-reindex-chainstate" if justchainstate else "-reindex", "-checkblockindex=1"])
+        while self.nodes[0].getblockcount() < blockcount:
+            time.sleep(0.1)
+        assert_equal(self.nodes[0].getblockcount(), blockcount)
         print("Success")
 
+    def run_test(self):
+        self.reindex(False)
```

Comment:
Just pass "-reindex-chainstate" or "-reindex" down instead of a bool. Bool parameters are an antipattern, plus this is just not clearer in any ways.